sdks/python/apache_beam/io/gcp/bigquery_tools.py

```diff
@@ -924,7 +924,8 @@ def get_or_create_table(
 
     # If table exists already then handle the semantics for WRITE_EMPTY and
     # WRITE_TRUNCATE write dispositions.
-    if found_table:
+    if found_table and write_disposition in (
+        BigQueryDisposition.WRITE_EMPTY, BigQueryDisposition.WRITE_TRUNCATE):
       table_empty = self._is_table_empty(project_id, dataset_id, table_id)
```

Comment:
I think you can limit calling `_is_table_empty` for only when the disposition is `WRITE_EMPTY`.

Reply:
good catch. LMK if it looks better now.